packages/view-types/heatmap/src/Heatmap.js

```diff
@@ -739,22 +746,6 @@ const Heatmap = forwardRef((props, deckRef) => {
       numCols: width,
     });
 
-    if (colI === null) {
-      if (transpose) {
-        setCellHighlight(null);
-      } else {
-        setGeneHighlight(null);
-      }
-    }
-
-    if (rowI === null) {
-      if (transpose) {
-        setGeneHighlight(null);
-      } else {
-        setCellHighlight(null);
-      }
-    }
-
```

Comment:
this code is not needed if you set the highlighted gene and the cell to null in the very beginning of the function and then add 
```
    if (obsId) {
      setCellHighlight(obsId);
    }
    if (varId) {
      setGeneHighlight(varId);
    }
```

Reply:
That is a good point. Can we use local mutable variables to keep track of the values and then call `setCellHighlight` and `setGeneHighlight` only once at the end of the `onHover` function?

Reply:
Of course, this is much better -- we don't want to cause a component that depends on them to rerender all the time! I have updated the code to use local variables and the setters are used only once at the end of the function.